cas/service.c

```diff
@@ -2610,6 +2679,26 @@ M0_INTERNAL int m0_cas_fom_spawn(
 	return M0_RC(rc);
 }
 
+M0_INTERNAL uint32_t m0_cas_svc_device_id_get(
+	const struct m0_reqh_service_type *stype,
+	const struct m0_reqh              *reqh)
+{
+	struct m0_reqh_service *svc;
+	struct cas_service     *cas_svc;
+
+	M0_PRE(stype != NULL);
+	M0_PRE(reqh != NULL);
+
+	svc = m0_reqh_service_find(stype, reqh);
+	M0_ASSERT(svc != NULL);
+	M0_ASSERT(m0_reqh_service_state_get(svc) == M0_RST_STARTED);
+
+	cas_svc = M0_AMB(cas_svc, svc, c_service);
+
+	M0_ASSERT(cas_svc->c_sdev_id != INVALID_CAS_SDEV_ID);
+	return cas_svc->c_sdev_id;
+}
+
```

Comment:
We would still need the changes which will confirm the sdev-id in the DIX FID is valid for the current CAS service just to make sure that no unwanted operations (because of code bugs) are serviced.

Reply:
For sure, but propose to add it as a separate feature to get current PR merged faster as it is quite critical. That could be quite a simple change in cas_device_check(), but possibly it may lead to some failures in unit tests where device id is not properly controlled.

Reply:
Yes as discussed in last evening's status meeting let us go ahead to merge this PR for now and the validation can be taken as a different PR after making sure all the UT's still work with the validation changes.